Predicate with range expression can't be encoded

If I create a predicate like this:

```
#Predicate<Dummy> {
    (0...10).contains($0.value)
}
```

And try to encode it with a `JSONEncoder`, I get this error:

`The 'Foundation.PredicateExpressions.RangeExpressionContains<Foundation.PredicateExpressions.ClosedRange<Foundation.PredicateExpressions.Value<Swift.Int>, Foundation.PredicateExpressions.Value<Swift.Int>>, Foundation.PredicateExpressions.KeyPath<Foundation.PredicateExpressions.Variable<PredicateRangeCoding.Dummy>, Swift.Int>>' type is not in the provided allowlist (required by /)`

Even though I've provided a codable configuration:

```
struct Dummy: PredicateCodableKeyPathProviding {
    static var predicateCodableKeyPaths: [String: PartialKeyPath<Dummy>] {
        ["value": \.value]
    }

    let value: Int
}

var configuration = PredicateCodableConfiguration.standardConfiguration
configuration.allowKeyPathsForPropertiesProvided(by: Dummy.self)
```

Encoding succeeds if I explicitly allow this type, like so:

```
configuration.allowType(PredicateExpressions.RangeExpressionContains<PredicateExpressions.ClosedRange<
                        PredicateExpressions.Value<Int>, PredicateExpressions.Value<Int>>,
                        PredicateExpressions.KeyPath<PredicateExpressions.Variable<Dummy>, Int>>.self)
```

More strangely, encoding also succeeds if I first encode a different predicate that does not contain a range, like this one:

```
#Predicate<Dummy> {
    $0.value > 0
}
```

Other types of predicates can be encoded without issue, so it seems specific to the range expression.

I've created a sample project here: https://github.com/ordo-one/external-reproducers/tree/main/swift/predicate-range-coding. I had a issue when building this on the command line, but it worked when building in Xcode, not sure why:

`main.swift:30:24: error: external macro implementation type 'FoundationMacros.PredicateMacro' could not be found for macro 'Predicate'`

Using Xcode 15.0 (15A240d) with Swift version 5.9 (swiftlang-5.9.0.128.108 clang-1500.0.40.1).

--- a/Sources/FoundationEssentials/Predicate/Archiving/PredicateCodableConfiguration.swift
+++ b/Sources/FoundationEssentials/Predicate/Archiving/PredicateCodableConfiguration.swift
@@ -352,6 +352,7 @@ extension PredicateCodableConfiguration {
         // Foundation-defined PredicateExpression types
         configuration.allowPartialType(PredicateExpressions.Arithmetic<PredicateExpressions.Value<Int>, PredicateExpressions.Value<Int>>.self, identifier: "PredicateExpressions.Arithmetic")
         configuration.allowPartialType(PredicateExpressions.ClosedRange<PredicateExpressions.Value<Int>, PredicateExpressions.Value<Int>>.self, identifier: "PredicateExpressions.ClosedRange")
+        configuration.allowPartialType(PredicateExpressions.RangeExpressionContains<PredicateExpressions.Value<Range<Int>>, PredicateExpressions.Value<Int>>.self, identifier: "PredicateExpressions.RangeExpressionContains")
         configuration.allowPartialType(PredicateExpressions.CollectionContainsCollection<PredicateExpressions.Value<[Int]>, PredicateExpressions.Value<[Int]>>.self, identifier: "PredicateExpressions.CollectionContainsCollection")
         configuration.allowPartialType(PredicateExpressions.CollectionIndexSubscript<PredicateExpressions.Value<[Int]>, PredicateExpressions.Value<Int>>.self, identifier: "PredicateExpressions.CollectionIndexSubscript")
         configuration.allowPartialType(PredicateExpressions.CollectionRangeSubscript<PredicateExpressions.Value<[Int]>, PredicateExpressions.Value<Range<Int>>>.self, identifier: "PredicateExpressions.CollectionRangeSubscript")

--- a/Tests/FoundationEssentialsTests/PredicateCodableTests.swift
+++ b/Tests/FoundationEssentialsTests/PredicateCodableTests.swift
@@ -331,6 +331,12 @@ final class PredicateCodableTests: XCTestCase {
         }
         decoded4 = try _encodeDecode(predicate4)
         XCTAssertEqual(try decoded4.evaluate(["A": 1, "B": 2, "C": 3]), try predicate4.evaluate(["A": 1, "B": 2, "C": 3]))
+        
+        let predicate5 = #Predicate<Int> {
+            (0 ..< 4).contains($0)
+        }
+        let decoded5 = try _encodeDecode(predicate5)
+        XCTAssertEqual(try decoded5.evaluate(2), try predicate5.evaluate(2))
     }
     
     func testMalformedData() {

